openmc/stats/multivariate.py

```diff
@@ -491,6 +491,121 @@ def from_xml_element(cls, elem):
         origin = [float(x) for x in elem.get('origin').split()]
         return cls(r, theta, phi, origin=origin)
 
+class CylindricalIndependent(Spatial):
+    """Spatial distribution represented in cylindrical coordinates.
+
+    This distribution allows one to specify coordinates whose :math:`r`,
+    :math:`\phi`, and :math:`z` components are sampled independently from
+    one another and centered on the coordinates (x0, y0, z0).
+
+    Parameters
+    ----------
+    r : openmc.stats.Univariate
+        Distribution of r-coordinates
+    phi : openmc.stats.Univariate
+        Distribution of phi-coordinates (azimuthal angle)
+    z : openmc.stats.Univariate
+        Distribution of z-coordinates
+    origin: Iterable of float, optional
+        coordinates (x0, y0, z0) of the center of the cylinder. Defaults to 
+        (0.0, 0.0, 0.0)
+
+    Attributes
+    ----------
+    r : openmc.stats.Univariate
+        Distribution of r-coordinates
+    phi : openmc.stats.Univariate
+        Distribution of phi-coordinates (azimuthal angle)
+    z : openmc.stats.Univariate
+        Distribution of z-coordinates
+    origin: Iterable of float, optional
+        coordinates (x0, y0, z0) of the center of the cylinder. Defaults to 
```

Comment:
The comment here (and elsewhere) doesn't actually match the sampling, where the z coordinate returned is the z0 + the sampled value. Was the intent to have z0 be the center of the cylinder, or just a reference point to which sampled values of z are added (what's currently done)?

Reply:
What I was trying to do was specify the origin of a new cylindrical reference frame from which the r, phi, z distributions are sampled and transformed to the global reference frame. I was trying to make sure the distribution parameters for the source refer to the same local reference frame rather than the global frame. I can ammend the parameter descriptions to specify that they are in a local frame shifted from the global one by the origin parameter. Now that I think about it more this implementation is limited to z-cylinders so it might be good to add a parameter for the axis direction of the cylinder in the global reference frame - it should be easy to do with the UnitSphere class I think. Thoughts?

Reply:
That's a good point. At the very least, the description of the origin parameter should be updated to more accurately describe what it's purpose is. I'll leave it up to you whether you want to add a parameter for the axis direction as part of this PR or if you'd prefer to leave it as a separate PR.

Reply:
Ah, I see you already fixed the description :+1: